File: frontend/src/model/project/types/project.ts
import type { Perspective } from "./perspective";

export type Project = {
  id: string;
  name: string;
  style: string;
  gameType: string;
  platform: string;
  description: string;
  reference: string;
};

export type ProjectSummary = Project & {
  perspective: Perspective;
  visualDirection: string;
  assetCount: number;
};

export type CreateProjectInput = Omit<ProjectSummary, "id" | "assetCount">;

File: core-api/internal/module/workspace/project/project.go
package project

type Project struct {
	UserID         uint
	ID             uint
	Name           string
	GameType       GameType    `json:"gameType"`
	Perspective    Perspective `json:"perspective"`
	TargetPlatform PlatformType
	Description    string
	Reference      string
	Style          string
}

// ProjectUpdate contains only fields supplied by a partial project update.
// A nil field is left unchanged; a non-nil field is written, including an empty value.
type ProjectUpdate struct {
	ID             uint
	Name           *string
	GameType       *GameType
	Perspective    *Perspective
	TargetPlatform *PlatformType
	Description    *string
	Reference      *string
	Style          *string
}
